src/content/guides/packages.md

```diff
@@ -178,6 +178,78 @@ due to conflicting version constraints in the pubspec.
 To identify out-of-date packages that require editing the pubspec,
 use [`dart pub outdated`][outdated].
 
+## Getting dependencies in production - `pub get --enforce-lockfile`
+
+In some cases `dart pub get` will not give you the exact versions locked in
+`pubspec.lock`. For example:
+
+* If new dependencies are added to `pubspec.yaml` since `pubspec.lock` was created.
+* If the locked version no longer exists in the package repository.
+* If you upgraded the Dart SDK, and some packages no longer are compatible with
+  the sdk.
+  
+In all these cases `dart pub get` will
+
+* Unlock as needed.
+* Notify about any changes in the output.
+
+For example after adding `retry: ^3.0.0` to your dependencies:
+```console
+$ dart pub get
+Resolving dependencies... (1.0s)
+Downloading packages... 
++ retry 3.1.2
+```
+
+Also if the hash of a published version does not match the hash in
+`pubspec.lock`, `pub` will warn you, but fix `pubspec.lock` to reflect the
+published version:
+
+For example if we manually change a hash in `pubspec.lock`:
+
+```console
+$ dart pub get
+Resolving dependencies... 
+Downloading packages... 
+~ retry 3.1.2 (was 3.1.2)
+The existing content-hash from pubspec.lock doesn't match contents for:
+ * retry-3.1.2 from "https://pub.dev"
+
+This indicates one of:
+ * The content has changed on the server since you created the pubspec.lock.
+ * The pubspec.lock has been corrupted.
+
+The content-hashes in pubspec.lock has been updated.
+
+For more information see:
+https://dart.dev/go/content-hashes
+Changed 1 dependency!
+```
+
+When deploying your project to production use `dart pub get --enforce-lockfile`.
+This will fail, if dependencies can't be satisfied with the exact versions and hashes in `pubspec.lock`. Thus, you avoid deploying untested dependencies to production.
```

Comment:
If possible, put the conditional clause first and be more specific about what "this" is. Would "package retrieval" be an appropriate replacement for "this" here, or perhaps "the command"?

Also, is it worth clarifying "dependencies and dependency versions"? As it seems both would be caught.

A possible alternative with some suggestions:

```suggestion
If your project's dependency constraints can't be satisfied with
the exact versions and content hashes in `pubspec.lock`,
package retrieval and the command will fail.
This helps avoid deploying untested
dependencies and dependency versions to production.
```